helm/templates/podmonitor.yaml

```diff
@@ -0,0 +1,53 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) ( .Values.podmonitor.enabled ) }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: {{ .Chart.Name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    monitoring: shared
+spec:
+  jobLabel: hypertrace-collector-pods
+  selector:
+    matchLabels:
+      {{- toYaml .Values.deploymentSelectorMatchLabels | nindent 6 }}
+  podMetricsEndpoints:
+  - interval: {{ .Values.podmonitor.interval }}
+    relabelings:
```

Comment:
does not need relabelings. you can directly specify the `port`. 
```
  podMetricsEndpoints:
  - port: {{ .Values.podMonitor.port }}
    interval: {{ .Values.podMonitor.interval }}
    scheme: {{ .Values.podMonitor.scheme }}
    {{- if .Values.podMonitor.tlsConfig }}
    tlsConfig:
      {{- .Values.podMonitor.tlsConfig | nindent 6 }}
    {{- end }}
```

Reply:
We cannot use a port that is not defined in deployments spec with a name like https://github.com/hypertrace/hypertrace-collector/blob/main/helm/templates/deployment.yaml#L64. This port is not defined and does not have a name. We can only use name for `port` in `podMetricsEndpoints`. 
https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#podmetricsendpoint

I tried locally, if we give port as "8889" it does not collect metrics. But it does when we use metric relabelings. Also, I added other relabelings as per https://github.com/Traceableai/traceable-monitoring/blob/master/helm/templates/kubernetes-pods-monitor.yaml so that these new metrics have similar labels as they currently have.

Reply:
We cannot use a port that is not defined in deployments spec with a name like https://github.com/hypertrace/hypertrace-collector/blob/main/helm/templates/deployment.yaml#L64. This port is not defined and does not have a name. We can only use name for port in podMetricsEndpoints.
https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#podmetricsendpoint

I tried locally, if we give port as "8889" it does not collect metrics. But it does when we use metric relabelings.

Reply:
Added containerPorts for these and used port in podmonitor